packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2680,6 +2680,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           return super.toAssignable(this.typeCastToParameter(node), isLHS);
         case "TSParameterProperty":
           return super.toAssignable(node, isLHS);
+        case "ParenthesizedExpression":
```

Comment:
Would doing this instead work?
```suggestion
        case "ParenthesizedExpression":
          if (node.expression.type === "TSAsExpression" || node.expression.type === "ParenthesizedExpression") {
            node.expression = this.toAssignable(node.expression, isLHS);
            return node;
          } else {
            return super.toAssignable(node, isLHS);
          }
```

Reply:
It does!